Adapter.C

```diff
@@ -923,6 +924,69 @@ void preciceAdapter::Adapter::setupCheckpointing()
+    {                                                                                                                 \
+        regFields.push_back(obj);                                                                                     \
+    }                                                                                                                 \
+    /* Iterate through checkpointed fields */                                                                         \
+    for (uint i = 0; i < GeomFieldCopies_.size(); i++)                                                                \
+    {                                                                                                                 \
+        found = false;                                                                                                \
+        obj = GeomFieldCopies_.at(i)->name();                                                                         \
+        for (uint j = 0; j < regFields.size(); j++)                                                                   \
+        {                                                                                                             \
+            if (obj == regFields.at(j))                                                                               \
+            {                                                                                                         \
+                found = true;                                                                                         \
+                break;                                                                                                \
+            }                                                                                                         \
+        }                                                                                                             \
+        if (!found)                                                                                                   \
+        {                                                                                                             \
+            toRemove.push_back(obj);                                                                                  \
+        }                                                                                                             \
+    }                                                                                                                 \
+    if (!toRemove.empty())                                                                                            \
+    {                                                                                                                 \
+        for (int i = toRemove.size() - 1; i >= 0; i--)                                                                \
```

Comment:
C++ has also reverse iterators

```suggestion
for (auto it = toRemove.rbegin(); it != toRemove.rend(); ++it)
```

Reply:
I actually need the index not the element. Then I would use std::distance, to get the index.

```C++
    if (!toRemove.empty())                                                                                            \
    {                                                                                                                 \
        for (auto it = toRemove.rbegin(); it != toRemove.rend(); ++it)                                                \
        {                                                                                                             \
            auto index = std::distance(toRemove.rbegin(), it);                                                        \
            GeomField_.erase(GeomField_.begin() + index);                                                             \
            delete GeomFieldCopies_.at(index);                                                                        \
            GeomFieldCopies_.erase(GeomFieldCopies_.begin() + index);                                                 \
            DEBUG(adapterInfo("Removed " #GeomField " : " + *it + " from the checkpointed fields list."));            \
        }                                                                                                             \
    }
```

Reply:
This is now implemented as in the comment.